### PR TITLE
NCF keras: Add custom loss and metrics 

### PR DESCRIPTION
**Pull request by guptapriya**
Added custom loss (using model.add_loss) and custom metrics (using custom layer) so that we can account for weights in those computations.
The results with this change seem similar to those with custom training loop:

```
Epoch 1/10
994/994 [==============================] - 46s 47ms/step - loss: 0.2469 - HR_METRIC: 0.0104 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.5334
Epoch 2/10
994/994 [==============================] - 29s 29ms/step - loss: 0.1873 - HR_METRIC: 0.0100 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.5875
Epoch 3/10
994/994 [==============================] - 29s 29ms/step - loss: 0.1689 - HR_METRIC: 0.0098 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6166
Epoch 4/10
994/994 [==============================] - 29s 29ms/step - loss: 0.1569 - HR_METRIC: 0.0101 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6203
Epoch 5/10
994/994 [==============================] - 27s 27ms/step - loss: 0.1487 - HR_METRIC: 0.0100 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6273
Epoch 6/10
994/994 [==============================] - 27s 27ms/step - loss: 0.1426 - HR_METRIC: 0.0103 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6305
Epoch 7/10
994/994 [==============================] - 26s 27ms/step - loss: 0.1383 - HR_METRIC: 0.0101 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6353
Epoch 8/10
994/994 [==============================] - 26s 27ms/step - loss: 0.1349 - HR_METRIC: 0.0106 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6347
Epoch 9/10
994/994 [==============================] - 27s 27ms/step - loss: 0.1322 - HR_METRIC: 0.0099 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6366
Epoch 10/10
994/994 [==============================] - 26s 27ms/step - loss: 0.1297 - HR_METRIC: 0.0098 - val_loss: 0.0000e+00 - val_HR_METRIC: 0.6361
```

Also removed the cloning flag as cloning is False by default now.

### CODE REVIEW

**Comment by guptapriya** [comment]
Closed. please review https://github.com/tensorflow/models/pull/6947 instead
